services/ux-backend/handlers/expandstorage/handler.go

```diff
@@ -0,0 +1,287 @@
+package expandstorage
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	storagecluster "github.com/red-hat-storage/ocs-operator/v4/controllers/storagecluster"
+	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, client client.Client, namespace string) {
+	switch r.Method {
+	case "POST":
+		handlePost(w, r, client, namespace)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(w http.ResponseWriter, r *http.Request, client client.Client, namespace string) {
+	// When ContentLength is 0 that means request body is empty and
+	var err error
+	if r.ContentLength == 0 {
+		klog.Errorf("body in the request is required")
+		http.Error(w, "body in the request is required", http.StatusBadRequest)
+		return
+	}
+
+	type poolDetails struct {
+		VolumeType           string `json:"volumeType"`
+		PoolName             string `json:"poolName"`
+		DataProtectionPolicy int    `json:"dataProtectionPolicy"`
+		EnableCompression    bool   `json:"enableCompression"`
+		FilesystemName       string `json:"filesystemName"`
+	}
+
+	type storageClassDetails struct {
+		ReclaimPolicy                string `json:"reclaimPolicy"`
+		Name                         string `json:"name"`
+		VolumeBindingMode            string `json:"volumeBindingMode"`
+		EnableStorageClassEncryption bool   `json:"enableStorageClassEncryption"`
```

Comment:
So far from UI it is mandatory to use KMS with StorageClass encryption. Here I can see `EnableStorageClassEncryption` is of type `bool`, so we are certain that we don't want KMS here ??

Will Ceph-CSI use K8s Secret instead to store keys ??

Reply:
We will use KMS itself the cluster-wide settings,
This field `EnableStorageClassEncryption` can set per device set encryption, whether a device set needs encryption or not,

Reply:
IIUC field `enableEncryption` is for deviceSet encryption and here we could use cluster-wide details.

but I am confirming about `enableStorageClassEncryption` field, which is for StorageClass encryption (not OSD/deviceSet), Ceph-CSI uses different ConfigMap than Rook for KMS.

Reply:
I am specifying this using bool setting  https://github.com/rook/rook/blob/master/deploy/examples/csi/rbd/storageclass.yaml#L51

can UI also provide https://github.com/rook/rook/blob/master/deploy/examples/csi/rbd/storageclass.yaml#L56?

Reply:
We need to display the same KMS form for this feature that UI currently displays on "Create StorageClass" page.

After that it's a design question that who will create/patch the csi-kms ConfigMap with KMS details. On StorageClass creation page (day-2) and during initial deployment (day-1) UI creates it... but here we are introducing new API endpoint, so whether this API should do it now or UI should do it and pass the `encryptionKMSID` in the payload.

Reply:
cc @vbnrh

Reply:
Added the field `encryptionKMSID`